src/apps/carbon-defi/ethereum/carbon-defi.strategy.contract-position-fetcher.ts

```diff
@@ -0,0 +1,142 @@
+import { Inject } from '@nestjs/common';
+import { BigNumberish } from 'ethers';
+import { sumBy } from 'lodash';
+
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { ETH_ADDR_ALIAS, ZERO_ADDRESS } from '~app-toolkit/constants/address';
+import { PositionTemplate } from '~app-toolkit/decorators/position-template.decorator';
+import { drillBalance } from '~app-toolkit/helpers/drill-balance.helper';
+import { ContractPositionBalance } from '~position/position-balance.interface';
+import { ContractPosition, MetaType } from '~position/position.interface';
+import { ContractPositionTemplatePositionFetcher } from '~position/template/contract-position.template.position-fetcher';
+import {
+  DefaultContractPositionDefinition,
+  UnderlyingTokenDefinition,
+  GetTokenBalancesParams,
+  GetTokenDefinitionsParams,
+  GetDisplayPropsParams,
+  GetDataPropsParams,
+} from '~position/template/contract-position.template.types';
+
+import { CarbonDefiViemContractFactory } from '../contracts';
+import { CarbonController } from '../contracts/viem/CarbonController';
+
+interface Order {
+  y: bigint;
+  z: bigint;
+  A: bigint;
+  B: bigint;
+}
+
+interface Strategy {
+  id: bigint;
+  owner: string;
+  tokens: readonly [string, string];
+  orders: readonly [Order, Order];
+}
+
+interface StrategyDefinition extends DefaultContractPositionDefinition {
+  strategy: Strategy;
+}
+
+type StrategyProps = {
+  strategy: Strategy;
+};
+
+function isActiveStrategy(strategy: Strategy) {
+  const [buy, sell] = strategy.orders;
+  return !!buy.A || !!buy.B || !!sell.A || !!sell.B;
+}
+
+@PositionTemplate()
+export class EthereumCarbonDefiStrategyContractPositionFetcher extends ContractPositionTemplatePositionFetcher<
+  CarbonController,
+  StrategyProps,
+  StrategyDefinition
+> {
+  groupId = 'strategy';
```

Comment:
Defining `groupId` isn't necessary because it's infered  from `@PositionTemplate()`. You can have a look at the implementation and you'll understand what I mean